20181010_Choropleth_Maps_in_R/Customisable Choropleth Maps.Rmd

```diff
@@ -0,0 +1,489 @@
+
+# Load in Shapefiles and Data
+
+We are going to be plotting England local authority data so will be using the corresponding shapefiles. We'll load them using the `rgdal` package. Notice we don't include the file extension:
+
+```{r, eval=TRUE}
+LAs <- rgdal::readOGR("Shapefiles","England_LAUA_2016") 
+```
+
+This loads in as a large spatial polygons data frame. We can use `plot()` to see what it looks like:
+```{r, eval=TRUE}
+plot(LAs)
+```
+
+Next we'll load in the data we'll be plotting. This is just some dummy data I created:
+```{r, eval=TRUE, message=FALSE}
+dummy <- read_csv("Data/LA_dummy_data.csv")
+
+head(dummy)
+```
+
+Now in order to plot the shapefile using ggplot2, we need to convert it from a spatial polygons data frame into a ggplot2-compatible data frame
+```{r, eval=TRUE,message=FALSE}
+tidy_LAs <- broom::tidy(LAs, region = "LAD16CD")
+
+head(tidy_LAs)
+```
+
+The region argument is the variable we are using to split up the areas. Here it is "LAD16CD" which is the Local Authority code. This is turned into an id variable.
+
+Now we will join our data with the converted shapefile so that we can plot it:
+```{r, message=FALSE}
+map_data <- tidy_LAs %>% 
+  left_join(dummy, by = c("id" = "LA_Code"))
+
+head(map_data)
+```
+
+# Plot the map
+
+Using `geom_polygon`, we'll plot the latitude against the longitude and then colour the map by the "Percentage" variable:
+
+```{r, eval=TRUE}
+map <- ggplot() + 
+  geom_polygon(data = map_data, 
+               aes(x = long, y = lat, group = group, fill = Percentage), 
+               col = "black")
+map
+```
+
+Since we're plotting a polygon with multiple rows of data, we need to overide the default 'group' and map it to a variable that has a different value for each group. In this case, when we converted our shapefile, a 'group' variable was created.
+
+Great, we have a map! We don't even need to define a colour scale as it's using the ggplot2 default continuous colour sclae. 
+
+It looks a bit distorted though. We can fix this by using `coord_equal()` to force a 1:1 aspect ratio
+```{r, eval=TRUE}
+map <- map + coord_equal()
+
+map
+```
+
+Okay, we have a good aspect ratio now, but we have a horrible grey background and axes that we don't really want. We can use the `theme()` function to set these things to blank:
+
+```{r, eval=TRUE}
+map + theme(panel.background = element_blank(), # get rid of grey background
+        axis.line = element_blank(), # get rid of axes lines
+        axis.title = element_blank(), # get rid of axes labels
+        axis.text = element_blank(), # get rid of the axes text
+        axis.ticks = element_blank()) # get rid of the tick marks
+```
+
+There are many elements we can use within the theme function. If there are standard things we want in all of our maps, we can create a function so that we don't need to call of these elements each time. We'll base this function on the built-in `theme_minimal()`:
+
+```{r, message=FALSE}
+theme_map <- function() {

```

Comment:
Any reason to wrap this up in a function? 

I think this works because the function returns the theme object, but as it doesn't take any params I wonder if its complicating things. 

This would do the same, I think: 
```r

theme_map <- 
  theme_minimal() +
    theme(
      text = element_text(family = "sans"), # Arial font
      axis.line = element_blank(),
      axis.text = element_blank(),
      axis.ticks = element_blank(),
      axis.title = element_blank(),
      panel.grid = element_blank(),
      plot.background = element_blank(), 
      panel.background = element_blank(), 
      legend.background = element_blank(),
      panel.border = element_blank()
    )

# Then no brackets needed for map + theme
map + theme_map 
```

Reply:
nope no reason! I'll change it